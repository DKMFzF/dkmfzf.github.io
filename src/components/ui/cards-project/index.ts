export { CardsProjectUI } from './cards-project';
