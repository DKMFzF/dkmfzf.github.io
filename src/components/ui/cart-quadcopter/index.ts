export { CartQuadcopter } from './cart-quadcopter';
