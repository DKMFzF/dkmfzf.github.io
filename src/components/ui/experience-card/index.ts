export { ExperienceCardUI } from './eperience-card';
