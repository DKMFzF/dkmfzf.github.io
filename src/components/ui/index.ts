export * from './preloader';
export * from './modal-overlay';
export * from './pages';
export * from './app-header';
export * from './modal';
export * from './icon';
export * from './link';
export * from './section-about-me';
