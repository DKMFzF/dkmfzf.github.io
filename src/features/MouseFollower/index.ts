export { MouseFollower } from './ui/MouseFollower';
