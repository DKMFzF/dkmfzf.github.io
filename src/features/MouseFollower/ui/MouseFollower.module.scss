.mouse-follower {
	border: 1px solid white;
	border-radius: 50%;
	height: 45px;
	left: 0;
	mix-blend-mode: difference;
	pointer-events: none;
	position: fixed;
	top: 0;
	transform: translate(-50%, -50%);
	width: 45px;
	z-index: 50;

	@media (width <= 1200px) {
		display: none;
	}
}
