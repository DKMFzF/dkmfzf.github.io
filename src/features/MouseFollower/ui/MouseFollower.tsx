import { useMouseFllower } from '../model/useMouseFllower';
import styles from './MouseFollower.module.scss';

export const MouseFollower = () => {
	const { circleRef } = useMouseFllower();

	return (
		<div
			ref={circleRef}
			className={styles['mouse-follower']}
			aria-hidden='true'
			role='presentation'
			data-testid='mouse-follower'
			data-cy='mouse-follower'
		/>
	);
};
