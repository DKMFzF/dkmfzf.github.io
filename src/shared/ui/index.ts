export * from './CommonPage';
export * from './Icon';
export * from './Preloader';
// export * from './Text';
