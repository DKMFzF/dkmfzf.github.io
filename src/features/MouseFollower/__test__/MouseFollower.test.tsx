import { MouseFollower } from '../ui/MouseFollower';
import { expect } from '@jest/globals';
import { render } from '@testing-library/react';

describe('[MouseFollowerUI]', () => {
	it('Renders mouse-follower correctly and matches snapshot', () => {
		const { asFragment } = render(<MouseFollower />);
		expect(asFragment()).toMatchSnapshot();
	});
});
