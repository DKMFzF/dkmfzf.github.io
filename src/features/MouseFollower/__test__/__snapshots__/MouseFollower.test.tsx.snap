// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[MouseFollowerUI] Renders mouse-follower correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="mouse-follower"
    data-cy="mouse-follower"
    data-testid="mouse-follower"
    role="presentation"
  />
</DocumentFragment>
`;
